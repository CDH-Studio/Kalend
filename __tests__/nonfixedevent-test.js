import React from 'react';
import renderer from 'react-test-renderer';
// import NonFixedEvent from '../src/components/screens/NonFixedEvent';
// import {Header} from 'react-navigation';
import {StatusBar, TouchableOpacity, StyleSheet, Text, Switch} from 'react-native';
// import Slider from '@react-native-community/slider';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';
//import DatePicker from 'react-native-datepicker';
import NumericInput from 'react-native-numeric-input';

// The test commented out gives erors in Jest since the time changes

// test("NonFixedEvent renders correctly", () => {
// 	const tree = renderer.create(<NonFixedEvent />).toJSON();
// 	expect(tree).toMatchSnapshot();
// });

test("StatusBar renders correctly", () => {
	const statusBar = renderer.create(<StatusBar translucent={true} backgroundColor={'#105dba'}/>).toJSON();
	expect(statusBar).toMatchSnapshot();
});

test('Icons render correctly', () => {
	const icon = renderer.create(<MaterialCommunityIcons name="face" size={130} color="#1473E6" />).toJSON();
	expect(icon).toMatchSnapshot();
});

test('Switch renders correctly', () => {
	const switchComponent = renderer.create(<Switch trackColor={{false: 'lightgray', true: '#FFBF69'}} ios_backgroundColor={'lightgray'} thumbColor={'#FF9F1C'} />).toJSON();
	expect(switchComponent).toMatchSnapshot();
});

// test('DatePickers render correctly', () => {
// 	const datePicker = renderer.create(<DatePicker showIcon={false} 
// 		mode="date" 
// 		style={{width:140}}
// 		customStyles={{dateInput:{borderWidth: 0}, dateText:{fontFamily: 'OpenSans-Regular'}, placeholderText:{color:'#565454'}}} 
// 		format="ddd., MMM DD, YYYY"
// 		confirmBtnText="Confirm" 
// 		cancelBtnText="Cancel" />).toJSON();
// 	expect(datePicker).toMatchSnapshot();
// });


test('NumericInputs render correctly', () => {
	const numInput = renderer.create(<NumericInput
		minValue={0} 
		leftButtonBackgroundColor={'#FFBF69'}
		rightButtonBackgroundColor={'#FF9F1C'}
		rounded={true}
		borderColor={'lightgray'}
		textColor={'#565454'}
		iconStyle={{ color: 'white' }}
		value = {0}
		onChange={(value) => {
			return value + 1;
		}} />).toJSON();
	expect(numInput).toMatchSnapshot();
});

// test('Slider renders correctly', () => {
// 	const slider = renderer.create(<Slider 
// 		minimumValue={0}
// 		maximumValue={1} 
// 		step={0.5}
// 		thumbTintColor={'#FF9F1C'}
// 		minimumTrackTintColor={'#FFBF69'}/>).toJSON();
// 	expect(slider).toMatchSnapshot();
// });

test("Buttons render correctly", () => {
	const button = renderer.create(<TouchableOpacity style={styles.buttonNext} onPress={() => this.props.navigation.navigate('NonFixedEvent')}>
	<Text style={styles.buttonNextText}>NEXT</Text>
	</TouchableOpacity>).toJSON();
	expect(button).toMatchSnapshot();
});

// const headerHeight = Header.HEIGHT;

const styles = StyleSheet.create({
	container: {
		flex: 1
	},

	// content: {
	// 	marginTop: StatusBar.currentHeight + headerHeight,
	// 	flex: 1,
	// 	paddingHorizontal: 15
	// },

	instruction: {
		flexDirection: 'row',
		justifyContent: 'center',
		alignItems: 'center'		
	},

	instructionText: {
		fontFamily: 'Raleway-Regular',
		color: '#565454',
		fontSize: 20,
		width: 205,
		marginLeft: 15
	},

	textInput: {
		flexDirection: 'row',
		alignItems: 'flex-end',
		marginRight: 5,
		height: 40
	},

	textInputBorder: {
		borderBottomColor: 'lightgray',
		borderBottomWidth: 1,
		width: '87%',
		marginLeft: 10,
	},

	sectionTitle: {
		fontFamily: 'Raleway-Medium',
		fontSize: 19,
		color: '#0A2239',
		marginBottom: 5,
		marginTop: 20
	},

	blueTitle: {
		color: '#1473E6',
		fontFamily: 'Raleway-SemiBold',
		fontSize: 17,
		width: 88
	},

	blueTitleLong: {
		color: '#1473E6',
		fontFamily: 'Raleway-SemiBold',
		fontSize: 17,
		width: 200
	},

	timeSection: {
		marginLeft: 10
	},

	dateRange: {
		flexDirection: 'row',
		justifyContent: 'space-between',
		alignItems: 'center',
		marginBottom: 5
	},

	questionLayout: {
		flexDirection: 'row',
		justifyContent: 'space-between',
		alignItems: 'center'
	},

	duration: {
		flexDirection: 'row',
		justifyContent: 'space-between',
		alignItems: 'center',
		marginBottom: 5
	},

	timePicker: {
		flexDirection: 'column',
		alignItems: 'center'
	},

	optionsText: {
		color: '#565454',
		fontFamily: 'OpenSans-Regular',
		marginBottom: 5
	},

	buttons: {
		flexDirection: 'row',
		justifyContent: 'center',
		alignItems: 'center',
		marginTop: 35
	},

	buttonEvent: {
		borderRadius: 12,
		backgroundColor: '#1473E6',
		width: 150,
		height: 57.9,
		elevation: 4,
		marginRight: 25,
		justifyContent:'center'
	},

	buttonEventText: {
		fontFamily: 'Raleway-SemiBold',
		fontSize: 15,
		color: '#FFFFFF',
		textAlign: 'center',
		padding: 8
	},

	buttonNext: {
		borderRadius: 12,
		backgroundColor: '#FFFFFF',
		width: 100,
		height: 58,
		borderWidth: 3,
		borderColor: '#1473E6',
		elevation: 4,
		justifyContent:'center'
	},

	buttonNextText: {
		fontFamily: 'Raleway-SemiBold',
		fontSize: 15,
		color: '#1473E6',
		textAlign: 'center',
		padding: 8
	}
});