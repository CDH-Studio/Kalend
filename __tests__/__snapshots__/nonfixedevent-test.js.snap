// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons render correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderColor": "#1473E6",
      "borderRadius": 12,
      "borderWidth": 3,
      "elevation": 4,
      "height": 58,
      "justifyContent": "center",
      "opacity": 1,
      "width": 100,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#1473E6",
        "fontFamily": "Raleway-SemiBold",
        "fontSize": 15,
        "padding": 8,
        "textAlign": "center",
      }
    }
  >
    NEXT
  </Text>
</View>
`;

exports[`Icons render correctly 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      Object {
        "color": "#1473E6",
        "fontSize": 130,
      },
      undefined,
      Object {
        "fontFamily": "Material Design Icons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;

exports[`NumericInputs render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      },
      Object {
        "borderColor": "lightgray",
        "height": 88,
        "width": 220,
      },
      Object {
        "borderRadius": 15.84,
      },
      Object {},
    ]
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFBF69",
        "borderBottomLeftRadius": 15.84,
        "borderTopLeftRadius": 15.84,
        "borderWidth": 0,
        "height": 86,
        "justifyContent": "center",
        "left": 0,
        "opacity": 1,
        "position": "absolute",
        "width": 66,
        "zIndex": -1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 33.44,
          },
          Array [
            Object {
              "backgroundColor": "rgba(0,0,0,0)",
              "fontWeight": "900",
            },
            Object {
              "color": "white",
            },
            Object {},
            Object {},
          ],
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "borderLeftColor": "lightgray",
          "borderLeftWidth": 1,
          "borderRightColor": "lightgray",
          "borderRightWidth": 1,
        },
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      editable={true}
      keyboardType="numeric"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      rejectResponderTermination={true}
      returnKeyType="done"
      style={
        Array [
          Object {
            "padding": 0,
            "textAlign": "center",
          },
          Object {
            "borderLeftColor": "lightgray",
            "borderLeftWidth": 1,
            "borderRightColor": "lightgray",
            "borderRightWidth": 1,
            "color": "#565454",
            "fontSize": 33.44,
            "height": 88,
            "width": 88,
          },
          Object {},
        ]
      }
      underlineColorAndroid="rgba(0,0,0,0)"
      value="0"
    />
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FF9F1C",
        "borderBottomRightRadius": 15.84,
        "borderTopRightRadius": 15.84,
        "borderWidth": 0,
        "height": 86,
        "justifyContent": "center",
        "opacity": 1,
        "position": "absolute",
        "right": 0,
        "width": 66,
        "zIndex": -1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 33.44,
          },
          Array [
            Object {
              "backgroundColor": "rgba(0,0,0,0)",
              "fontWeight": "900",
            },
            Object {
              "color": "white",
            },
            Object {},
            Object {},
          ],
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`Slider renders correctly 1`] = `
<RCTSlider
  disabled={false}
  enabled={true}
  maximumValue={1}
  minimumTrackTintColor="#FFBF69"
  minimumValue={0}
  onChange={null}
  onResponderTerminationRequest={[Function]}
  onSlidingComplete={null}
  onStartShouldSetResponder={[Function]}
  onValueChange={null}
  step={0.5}
  style={
    Object {
      "height": 40,
    }
  }
  thumbTintColor="#FF9F1C"
  value={0}
/>
`;

exports[`StatusBar renders correctly 1`] = `null`;

exports[`Switch renders correctly 1`] = `
<RCTSwitch
  accessibilityRole="button"
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onTintColor="#FFBF69"
  style={
    Array [
      Object {
        "height": 31,
        "width": 51,
      },
      Object {
        "backgroundColor": "lightgray",
        "borderRadius": 16,
      },
    ]
  }
  thumbTintColor="#FF9F1C"
  tintColor="lightgray"
  value={false}
/>
`;
